lib/server.js

```diff
@@ -168,3 +170,30 @@ exports.configure = function (config) {
   }
 }
 
+// calculate disk usage in hosting directory
+exports.updateDiskUsage = function(config) {
+  return new Promise(resolve => {
+    getFolderSize(config.directory, (err, size) => {
+      if (err) {
+        throw err
+      }
+      // - `size` is in bytes
+      // - when we fully parse the tree our "estimate" is exactly correct
+      // - this will drift over time as blocks come in
+      // - we keep a running estimate so we don't need to parse the full
+      //   storage folder on every block that is downloaded
+      config.diskUsage = size
+      config.estimatedDiskUsage = size
+
+      // if we are under quota, restart all dats
+      // is this the right place to do this?? config seems wrong but when we
+      // update the disk space seems right
+      console.log(config.diskUsage, config.diskQuota, config.diskUsage/config.diskQuota)
+      if (config.diskUsage < config.diskQuota) {
+        config.dats.forEach(datCfg => dat.start(datCfg, config))
```

Comment:
This isn't the correct usage of `dat.start`. See https://github.com/beakerbrowser/homebase/blob/master/lib/server.js#L141 -- the method returns a server instance which is supposed to be integrated into the top level express server.